docs/hyperion-setup.md

```diff
@@ -1 +1,411 @@
+
+```yaml
+database-url: "postgresql://localhost/postgres?user=postgres&password=mysecretpassword"
+port: 8081
+granularity: 10 # 10 seconds
+aggregation-ttl: 604800 # 7 days
+
+pipeline:
+    manager-host: "localhost:3000"
+    plugin-id: "Aggregator"
+```
+
+With this configuration, we will be storing an intermediate sum of logs that occurred every 10 seconds, with those entries expiring after 7 days. That means that the minimum granularity we will be able to query from the IDE will be 10 seconds, and that metrics may lag up to 10 seconds behind. It goes without saying that decreasing this value will allow for more granular querying, at the cost of more storage required.
+
+Finally, add the aggregator to the list of plugins in the plugin manager config. Your final plugin manager configuration should now look something like this:
+
+```yaml
+host: "tcp://localhost:3000"
+plugins:
+  - id: "ElasticSearch"
+    host: "tcp://localhost:3001"
+  - id: "Renamer"
+    host: "tcp://localhost:3002"
+  - id: "PathExtractor"
+    host: "tcp://localhost:3003"
+  - id: "Adder"
+    host: "tcp://localhost:3004"
+  - id: "VersionTracker"
+    host: "tcp://localhost:3005"
+  - id: "Aggregator"
+    host: "tcp://localhost:3006"
+```
+
+Launching the aggregator works the same as launching plugins:
+
+```shell script
+$ java -jar path/to/aggregator-all.jar aggregator.yml
+```
+
+That's it! Your log messages should now be going from ES, through the pipeline, into the aggregator. You can confirm whether or not your aggregator is working by requesting the following URL:
+
+```
+http://localhost:8081/api/v1/metrics?project=mock-logging&file=src/main/java/com/sap/enterprises/server/impl/TransportationService.java&intervals=10,60
+```
+
+It should return a list of metrics encountered in the TransportationService file in the last 10 seconds and minute.
+
+### Step 7: Intellij Setup
+
+Now that we have the data in our aggregator, it is time to actually use it.
+
+TODO: Plugin setup.
+
+### Wrapping Up
+
+So what have we actually created? Let's revisit the diagram from the beginning of this document, but with all of the plugins explicitly laid out.
+
+![](https://i.imgur.com/cJRicZi.png)
+
+This may seem like a complex setup, but the protocol that underlies the plugins is very optimized and allows thousands of messages handled per second on commodity hardware. By using a pipeline setup you can customize Hyperion for your own usecases, and even [write your own plugin](/docs/writing-java-kotlin-plugin.md) if needed.
+
+The [example](/example/) folder in this repository contains the files used in this tutorial. If you want to refer to the final working setup, you can reference that folder.
+
+Want more information on how to use the default pipeline plugins bundled with Hyperion to configure your own pipeline? Check out the [example pipeline configurations](/doc/advanced-examples.md) documentation article for a list of plugins and the situations in which they can be used.
```

Comment:
Link is incorrect should be (doc -> docs):
`[example pipeline configurations](/doc/advanced-examples.md) -> [example pipeline configurations](/docs/advanced-examples.md)`